File: Demos/Datasnap/Server/proxy/java_blackberry/com/embarcadero/javablackberry/TDataSet.java
//*******************************************************
//
//               Delphi DataSnap Framework
//
// Copyright(c) 1995-2011 Embarcadero Technologies, Inc.
//
//*******************************************************

package com.embarcadero.javablackberry;

/**
 * 
 * TDataSet is the base class for all dataset components that represent data in rows and columns.
 *
 */

public class TDataSet extends TDBXReader {
	public TDataSet(TParams params, TJSONObject value) {
		super(params, value);
	}

	/**
	 *  Returns a TDataSet created by the information contained in the JSONObject
	 * 
	 * @param value a JSONObject that contains the parameters for create the TDataSet  
	 * @return return the TDataSet object created
	 * @throws DBXException
	 */
	
	public static TDataSet CreateFrom(TJSONObject value) throws DBXException {
		TParams params = TParams.CreateParametersFromMetadata(value
				.getJSONArray("table"));
		TDataSet dst = new TDataSet(params, value);
		return dst;
	}
}

File: Demos/Datasnap/Server/proxy/objectivec_ios81/Proxy/TDataSet.m
//*******************************************************
//
//               Delphi DataSnap Framework
//
// Copyright(c) 1995-2011 Embarcadero Technologies, Inc.
//
//*******************************************************

#import "TDataSet.h"


@implementation TDataSet

@end

@implementation TDataSet (TDatasetCreation)

+(id) DataSetWithJSON:(TJSONObject *) value {
	TParams * params = [TParams createParametersWithMetadata:[value getJSONArrayForKey:@"table" ]] ;
	
	return [ TDBXReader DBXReaderWithParams:params andJSONObject: value];
}
+(id) DataSetWithParams: (TParams * )params andJSONObject: (TJSONObject *)json{
	return [[TDataSet alloc]initWithParams:params andJSONObject:json];
}
@end
